File: 21.merge_two_sorted_lists.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        if list1 is None:
            return list2
        if list2 is None:
            return list1

        
        head = list1 if list1.val < list2.val else list2
        curr = head
        if curr == list1:
            list1 = list1.next
        elif curr == list2:
            list2 = list2.next

        while list1 is not None and list2 is not None:
            if list1.val < list2.val:
                curr.next = list1
                curr = curr.next
                list1 = list1.next
            else:
                curr.next = list2
                curr = curr.next
                list2 = list2.next
        curr.next = list1 if list2 is None else list2
        return head

File: pres/21.merge_two_sorted_lists.c
/** Definition for singly-linked list. */
#include <stdio.h>
#include <stdlib.h>

struct ListNode {
    int val;
    struct ListNode *next;
};

#define MIN(a, b) ((a->val > b->val) ? a : b)

struct ListNode* mergeTwoLists(struct ListNode* list1, struct ListNode* list2){
    if (list1 == NULL)
        return list2;
    if (list2 == NULL)
        return list1;

    struct ListNode *head = malloc(sizeof(*head));
    while (1) {
    }
}

void add_min()
